core/string/string_buffer.h

```diff
@@ -35,7 +35,7 @@
 
 template <int SHORT_BUFFER_SIZE = 64>
 class StringBuffer {
-	char32_t short_buffer[SHORT_BUFFER_SIZE];
+	char32_t short_buffer[SHORT_BUFFER_SIZE] = {};
```

Comment:
I don't know the `StringBuffer` class, but I would double check this is required.

This is kind of controversial - it might lead to needless extra processing if used in a bottleneck, if not optimized out. Profiling / benchmarking may help, maybe there's no cost in practice but it would be nice to show this first. :man_shrugging: 

Some people often do this kind of thing thinking it is somehow "safer", but this is an illusion. Providing you aren't reading from uninitialized data there's no problem. Personally if it's a small structure and not a bottleneck I'd be more likely to zero. If a bottleneck, no. If a large buffer no (because zeroing commits the pages).